modules/beta-private-cluster/variables.tf

```diff
@@ -721,3 +720,45 @@ variable "enable_identity_service" {
   description = "Enable the Identity Service component, which allows customers to use external identity providers with the K8S API."
   default     = false
 }
+
+variable "strategy" {
+  type        = string
+  description = "The upgrade stragey to be used for upgrading the nodes. Valid values of state are: `SURGE`; `BLUE_GREEN`. By default strategy is `SURGE` (Optional)"
+  default     = "SURGE"
+}
+
+variable "max_surge" {
+  type        = number
+  description = "The number of additional nodes that can be added to the node pool during an upgrade. Increasing max_surge raises the number of nodes that can be upgraded simultaneously. Can be set to 0 or greater (Optional)"
+  default     = null
+}
+
+variable "max_unavailable" {
+  type        = number
+  description = "The number of nodes that can be simultaneously unavailable during an upgrade. Increasing max_unavailable raises the number of nodes that can be upgraded in parallel. Can be set to 0 or greater (Optional)"
+  default     = null
+}
+
+variable "node_pool_soak_duration" {
```

Comment:
Aren't these keys within individual nodepool objects within `node_pools` var? I don't think we need these here.
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/f397f5f35ce5a77392e9032c18193f4112907aa7/modules/beta-private-cluster/variables.tf#L145C10-L154

Reply:
Probably not. I will remove it from variables. :+1:

Reply:
However, I was follow the same declarations to `max_surge` and `max_unavailable`. These values can be individual values by nodepool but the are declared in variables. Anyway, I will remove blue green variables from file.

Reply:
Done. :heavy_check_mark: